### PR TITLE
Storages: Refine metrics of read threads and data sharing

### PR DESCRIPTION
**Pull request by JinheLin**
### What problem does this PR solve?

Issue Number: ref #8652

### What is changed and how it works?
- Add CPU usage metrics of `SegmentReader`.
image

- Refine metrics of read thread and data sharing.
image
image
image

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/SegmentReadTaskPool.cpp [hunk 1]
```diff
@@ -279,6 +279,7 @@ void SegmentReadTaskPool::pushBlock(Block && block)
     blk_stat.push(block);
     global_blk_stat.push(block);
     read_bytes_after_last_check += block.bytes();
+    GET_METRIC(tiflash_storage_read_thread_counter, type_push_block).Increment();
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/SegmentReadTaskPool.cpp:282]: Maybe the "bytes"/"rows" is more meaningful than the number of blocks?

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**